bin/comment_with_dependency_diff

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+set -euo pipefail
+
+MODULE="${1:-}"
+CONFIGURATION="${2:-}"
+
+if [ -z "$MODULE" ] || [ -z "$CONFIGURATION" ]; then
+  echo "Not enough arguments provided. Usage: ./comment_with_dependency_diff <MODULE> <CONFIGURATION>"
+  exit 1
+fi
+
+OWNER=$(basename "$(dirname "${BUILDKITE_PULL_REQUEST_REPO}")")
+REPO=$(basename "${BUILDKITE_PULL_REQUEST_REPO%.git}")
+
+DIFF_DEPENDENCIES_FOLDER="./build/reports/diff"
+BASE_BRANCH_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/base_branch_dependencies.txt"
+HEAD_BRANCH_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/head_branch_dependencies.txt"
+BASE_BRANCH_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/base_branch_build_env.txt"
+HEAD_BRANCH_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/head_branch_build_env.txt"
+DIFF_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/diff_dependencies.txt"
+DIFF_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/diff_build_env.txt"
+DEPENDENCY_TREE_VERSION="1.2.1"
+
+PR_NUMBER=$BUILDKITE_PULL_REQUEST
+PR_URL="https://api.github.com/repos/$OWNER/$REPO/pulls/$PR_NUMBER"
+
+git config --global user.email "mobile@automattic.com"
+git config --global user.name "Dependency Tree Diff Tool"
+
+echo "--> Starting the check"
+echo "--> Fetching the base branch from $PR_URL"
+githubResponse="$(curl "$PR_URL" -H "Authorization: token $GITHUB_TOKEN")"
```

Comment:
💭  I have not checked if this would be strictly equivalent or not, but what do you think about using [our `github_api` helper from this repo](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/blob/trunk/bin/github_api) instead of calling `curl` directly, now that this script lives in the same repo/plugin? 🤔